### ISSUE TITLE
TWE Design System: Code Drop Down component

### ISSUE THREAD

**Issue by joshfishman** [hackforla/internship#486]
### Overview
We need to create Design System Select Component HTML and SASS / connect Figma design to finished code, and build microsite page so that developers can see and easily change component examples, states, code and location.

### Dependencies
- #543 

#### Assumptions
- You have read the [CONTRIBUTING.md](https://github.com/hackforla/internship-website-design-system/blob/main/CONTRIBUTING.md) in the Design System repo
  - You have set up your development environment for working with the Internship Design System Repository: https://github.com/hackforla/internship-website-design-system
  - You will continue to follow the instructions in CONTRIBUTING.md file

### Action Items
- [x] Review the Microsite page mockup you will be working from (Resource 1.01)
- [x] Name the Branch you create Select
- [x] Determine if any JS or CSS hacks need to be included to style this component (see Resource 1.02 for examples of prior hacks).  If you need to use a hack, don't forget to update the examples.
- [x] Include necessary files in the `/docs` folder (Resource 1.03) and project config file `mkdocs.yml` (Resource 1.04)
- [x] Determine what variables are needed for this component and add them to variables.scss (Resource 1.05)
- [x] Code Microsite component page
   - file locations
     - [x] md file goes into the /docs folder (Resource 1.03) 
     - [x] SASS file goes into the /docs/components/sass/components folder (Resource 1.06)
   - [x] Create variables in sass sheet as necessary, add color varialbles with name determined by HTML Color Codes (Resource 1.07, 1.08)
   - [x] Create component classes and states within the page
   - [x] Include Files via code snippets to create the code section (see Resource 1.09 for details)
   - [x] Add links to SASS and Markdown files below the code (see examples in Resource 1.10)
- [x] Review template frontend for errors 
- [x] Make a PR
- Once PR is merged
   - [x] connect component in Figma to GitHub Files (Resource 1.10)
   - [x] Add links to `Resources gathered during the completion of this issue` section
      - [x] 2.01 MD file(s) created
      - [x] 2.02 SASS file(s) created
      - [x] 2.03 link to Header Microsite page on live site: If the site is live when this issue is worked on.

#### Resources for working on this issue
- 1.01 Design Reference
   -  [Figma, Select Microsite page](https://www.figma.com/design/TTRS2FWXsrymHYpPJL1IdH/Internship-Team-Main-file?node-id=2-54&p=f&t=d9p36VpyFreC0BBo-0)
   - [Design Systems Figma](https://www.figma.com/file/TTRS2FWXsrymHYpPJL1IdH/Internship-Team-Main-file?type=design&node-id=2%3A45&mode=design&t=jgMN8QdoLnh9F7MT-1)
- 1.02 WIKI: [Design system microsite JavaScript and sass hacks](https://github.com/hackforla/internship/wiki/Design-system-microsite-javascript-and-sass-hacks.md)
- 1.03 MKdocs folder: [/docs](https://github.com/hackforla/internship-website-design-system/tree/main/docs) on repo
- 1.04 MDdocs config file: [mkdocs.yml](https://github.com/hackforla/internship-website-design-system/blob/main/mkdocs.yml) on repo
- 1.05 variables file: [/docs/components/sass/abstracts/_variables.scss](https://github.com/hackforla/internship-website-design-system/tree/main/docs/components/sass/abstracts) on repo
- 1.06 SASS file folder: [/docs/components/sass/components](https://github.com/hackforla/internship-website-design-system/tree/main/docs/components/sass/components)
- 1.07 Variables SASS Sheet: [/docs/components/sass/abstracts/variables.sss](https://github.com/hackforla/internship-website-design-system/tree/main/docs/components/sass/abstracts/variables.scss)
- 1.08 HTML Color Code Names: [HTMLColorCodes.com](https://htmlcolorcodes.com)
- 1.09 WIKI: [How to Create a Design System Microsite Page using Snippets](https://github.com/hackforla/internship/wiki/How-to-Create-a-Design-System-Microsite-Page-using-Snippets)
- 1.10 e.g., 
   - below the CSS tab, it would say CSS: https://github.com/hackforla/internship-website-design-system/blob/main/docs/components/sass/components/_Select.scss
   - below the HTML tab it would say HTML: https://github.com/hackforla/internship-website-design-system/blob/main/docs/Select.md
- 1.11 [Connect component in Figma to GitHub Files](https://github.com/hackforla/internship/wiki/Connect-component-in-Figma-to-GitHub-Files)
- 1.12 Original Design Issue:#486
   - Microsite Epic Issue with list of design issues: #https://github.com/hackforla/internship/issues/455#issuecomment-2057437256

#### Resources gathered during the completion of this issue
- 2.01 MD file(s) created: [dropdowns.md] - https://github.com/hackforla/internship-website-design-system/blob/main/docs/dropdowns.md
- 2.02 SASS file(s) created: [_dropdowns.scss] - https://github.com/hackforla/internship-website-design-system/blob/main/docs/components/sass/components/_dropdowns.scss
- 2.03 link to Header Microsite page on live site: https://hackforla.github.io/internship-website-design-system/dropdowns/

**Comment by jnomad21** [comment]
Completed coding for dropdowns.  Need review by Dev Lead.  Have some questions for design that I will touch on during our meeting.

**Comment by jnomad21** [comment]
Updated formatting to reflect classes and states per specs discussed in the meeting.  Still need to update the displayed code per the new import process.  Have not reviewed that process as of yet.

**Comment by jnomad21** [comment]
Discussed snippet content yesterday.  Still need to update code per specs.

**Comment by Brayheart** [comment]
Hello @joshfishman, I've successfully pulled the internship design system repository and have it running locally. I've also reviewed the dropdown input field in the Figma design. Could you guide me on the subsequent actions I should take?

**Comment by ExperimentsInHonesty** [comment]
@Brayheart We are moving this issue back into the icebox because it has a new dependency.  After that dependency is resolved, it will be available for assignment again.  In the meantime, Joshua will assign you a new issue.  Sorry for the confusion.

**Comment by ExperimentsInHonesty** [comment]
@Thinking-Panda 
Please provide update
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures or links* (if necessary): "Add any pictures or links that will help illustrate what you are working on."

* remember to add links to the top of the issue if they are going to be needed again.

**Comment by Thinking-Panda** [comment]
Progress: Analyzed and understood the structure and files in the repo.
Blocker: Having trouble setting up docker to live reload.
Availability: Weekdays 10-3PM

**Comment by revforyou** [comment]
Approved by Bonnie in the Agenda meeting, merged after confirmation with Joshua, closing the Issue now.
